include/internal/catch_tostring.cpp

```diff
@@ -214,7 +214,7 @@ std::string StringMaker<char>::convert(char value) {
         return "'\\n'";
     } else if (value == '\t') {
         return "'\\t'";
-    } else if ('\0' <= value && value < ' ') {
+    } else if (value < ' ') {
```

Comment:
This changes the semantics, whether `char` is signed is implementation defined. In some platforms their ABI may indicate `char` is unsigned, some others signed: In Visual C++ char is signed by default, in GCC unsigned...
I think the fix is to use this instead:
`if(unsigned(value) < ' ')`